includes/admin/load.php

```diff
@@ -375,3 +394,35 @@ function addPluginProgressIndicator( message ) {
 		array( 'type' => 'module' )
 	);
 }
+
+/**
+ * Callback function that returns plugin settings link.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $plugin_slug Plugin slug passed to generate the settings link.
+ * @return string|null Either the plugin settings link or null if not available.
+ */
+function perflab_get_plugin_settings_link( string $plugin_slug ): ?string {
+	$plugin_file = null;
+
+	foreach ( array_keys( get_plugins() ) as $file ) {
+		if ( strtok( $file, '/' ) === $plugin_slug ) {
+			$plugin_file = $file;
+			break;
+		}
+	}
+
+	if ( null === $plugin_file ) {
+		return null;
+	}
+
+	/** This filter is documented in wp-admin/includes/class-wp-plugins-list-table.php */
+	$plugin_links = apply_filters( "plugin_action_links_{$plugin_file}", array() );
+
+	if ( is_array( $plugin_links ) && array_key_exists( 'settings', $plugin_links ) ) {
+		return $plugin_links['settings'];
+	}
```

Comment:
Per https://github.com/WordPress/performance/pull/1208/files#r1603609835:

```suggestion
	if ( ! is_array( $plugin_links ) || ! array_key_exists( 'settings', $plugin_links ) ) {
		return null;
	}
	
	$p = new WP_HTML_Tag_Processor( $plugin_links['settings'] );
	if ( ! $p->next_tag( array( 'tag_name' => 'A' ) ) ) {
		return null;
	}
	$href = $p->get_attribute( 'href' );
	if ( $href && is_string( $href ) ) {
		return $href;
	}
```